### DISCUSSION TITLE
It is possible to store data on the Android Webview Window Object at start with Capacitor/Ionic?

### DISCUSSION THREAD

**Discussion by distante** [ionic-team/capacitor#7384] [Q&A] [answered]
I have some configuration variables that depend on which flavor the Android App is using, including some auth variables that are needed for the angular app start up.

I need to write some of that data into a global const in the window object, but I do not find any way to do it.

I have tried creating a Capacitor Plugin and calling

```kotlin
override fun load() {
    this.bridge.webView.evaluateJavascript("const __myTest = 'hello world'", null);
}
```

or with

```kotlin
override fun handleOnStart() {
  super.handleOnStart()
  this.bridge.webView.evaluateJavascript("const saninn = 'hello saninn'", null);
}
```

But even when that part is called, it does not set anything (probably the webview is not defined at that time)

Is this even possible? I do not want to define a plugin call because I would need to change all the initialization of the app and I am trying to prevent that.

I am assuming there should be a way to do this because at the start the `Capacitor` object already exists on `window` .

**Comment by jcesarmobile** [comment]
It should be possible, but you can't create it as a `const` if you want it to be available globally, should be an object on `window` object, something like `window.saninn = 'hello saninn'`

**Reply by distante** [reply]
Sadly no, it is still not defined at start :(

**Reply by jcesarmobile** [reply, marked as answer]
Looks like some threading issue, it works if you use `this.bridge.eval("window.saninn  = 'hello saninn';", null);`, which runs `evaluateJavascript` under the hood, but on main thread.

**Reply by distante** [reply]
I will mark it as an answer, although I went in another route and loaded a Plugin with the flavor name as workaround.
